service/src/test/java/uk/gov/hmcts/reform/fpl/service/respondent/RespondentAfterSubmissionValidatorTest.java

```diff
@@ -1,23 +1,36 @@
 package uk.gov.hmcts.reform.fpl.service.respondent;
 
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.NullAndEmptySource;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
 import uk.gov.hmcts.reform.ccd.model.Organisation;
 import uk.gov.hmcts.reform.fpl.model.CaseData;
 import uk.gov.hmcts.reform.fpl.model.Respondent;
 import uk.gov.hmcts.reform.fpl.model.RespondentSolicitor;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.service.FeatureToggleService;
+import uk.gov.hmcts.reform.fpl.service.UserService;
 
 import java.util.List;
 import java.util.UUID;
+import java.util.stream.Stream;
 
-import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.mock;
-import static uk.gov.hmcts.reform.fpl.enums.YesNo.NO;
+import static org.mockito.quality.Strictness.LENIENT;
 import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
 import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
 
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = LENIENT)
```

Comment:
how come you're adding `LENIENT` strictness here? is it definitely needed at the class level? would it make sense to limit it to any individual methods that might need it, while retaining the benefits of stricter stubbing in general?

Reply:
the problem is that it should be applicable on most of test methods, especially ones that are parametrised. If we have parametrised test has 4 different combinations like in out cases toggle and user role, one one of combination does not use mock then you need to add @MockitoSettings(strictness = LENIENT) on a method or write separate test method.

E.g problem is with if(toggle.isNoCEnabled && userService.isAdmin). when toggle.isNoCEnabled is false then userService.isAdmin is not executed at all and we have unnecesary stubbing warning.
What if we change order  
if(userService.isAdmin && toggle.isNoCEnabled)
if you dont have @MockitoSettings(strictness = LENIENT)  than this will fail as now if userService is false then  isNoCEnabled in not executed at all. But these conditions are logically same and test should pass, but it wont due to unnecessary stubbing. Thats is why i dont like this warning, even if code is same logically your test will start failing.

Dont want to put annotation on method also, because test should be as clean as possible, and focus on scenario and not distract you with not important things

Reply:
regarding example with changing order, it may be different when performance is important factor, if e.g toggle is fast but user.isAdmin is slow, then it is better to have toggle condition first, but performance consideration should not have massive impact on functional tests